### PR TITLE
feat(Modal): create TwoScreenModal variant

### PR DESCRIPTION
**Pull request by natalieweesh**
### Overview

create new TwoScreenModal variant

the /account/billing page in the monolith uses two modals that have an initial screen with a continue button, then a confirm screen within the same modal. this variant is to make sure this two screen modal pattern is more accessible, in that when you click the continue button, the x button at the top right of the modal should be selected and the new title of the modal should be announced.

for example, the change plan modal:
Screen Shot 2022-07-05 at 2 13 47 PMScreen Shot 2022-07-05 at 2 13 52 PM

BEFORE: ☹️ 
ezgif-3-516c072401
when you click Continue button to go to the confirm screen of the modal, the Upgrade button at the bottom of the modal is highlighted and the new modal title is not announced

NOW: 😄 
ezgif-3-b0883af41b
when you click the Continue button to go to the confirm screen of the modal, the new modal title is announced and the X button at the top is highlighted

### PR Checklist

- [ ] Related to designs: https://www.figma.com/file/DeETaHGvA6uAXDZwGyjCFH/Upgrade-Downgrade-Pause
- [ ] Related to JIRA ticket: EGG-2633
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] This PR includes testing instructions tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

[Don't make me tap the sign.](https://i.imgur.com/sy93D9I.png)

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/31477)  | [Monolith Env](https://pr-31477-monolith.dev-eks.codecademy.com/) |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.153-alpha.47a396.0<br/>@codecademy/gamut-labs@32.4.1-alpha.47a396.0<br/>@codecademy/gamut@48.1.0-alpha.47a396.0<br/>@codecademy/monaco@0.6.56-alpha.47a396.0<br/>@codecademy/styleguide@58.2.0-alpha.47a396.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62c480e2c3fe5608fb82f1cd--gamut-preview.netlify.app>https://62c480e2c3fe5608fb82f1cd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62c480e2c3fe5608fb82f1cd>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.153-alpha.2c149b.0<br/>@codecademy/gamut-labs@32.4.1-alpha.2c149b.0<br/>@codecademy/gamut@48.1.0-alpha.2c149b.0<br/>@codecademy/monaco@0.6.56-alpha.2c149b.0<br/>@codecademy/styleguide@58.2.0-alpha.2c149b.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.153-alpha.713bde.0<br/>@codecademy/gamut-labs@32.4.1-alpha.713bde.0<br/>@codecademy/gamut@48.1.0-alpha.713bde.0<br/>@codecademy/monaco@0.6.56-alpha.713bde.0<br/>@codecademy/styleguide@58.2.0-alpha.713bde.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62c48e19c1a03d18a0d03d2e--gamut-preview.netlify.app>https://62c48e19c1a03d18a0d03d2e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62c48e19c1a03d18a0d03d2e>Deploy Logs</a></p>

**Review by dreamwasp — COMMENTED** [review]
I like the idea of this - couple notes!
* I think this should be more flexible - maybe a multiscreen modal that takes an array of views.
* I think it would make more sense for this to be accessed through our OG Modal component instead of a new component with a new name. The access pattern I'm thinking of is something like:

```
<Modal views={arrayOfNodes (or objects)} />
```

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.153-alpha.1c0e5e.0<br/>@codecademy/gamut-labs@32.4.1-alpha.1c0e5e.0<br/>@codecademy/gamut@48.1.0-alpha.1c0e5e.0<br/>@codecademy/monaco@0.6.56-alpha.1c0e5e.0<br/>@codecademy/styleguide@58.2.0-alpha.1c0e5e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62c5964f7a1b98008f2b68ee--gamut-preview.netlify.app>https://62c5964f7a1b98008f2b68ee--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62c5964f7a1b98008f2b68ee>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62c5a4a68cc4580fe187a806--gamut-preview.netlify.app>https://62c5a4a68cc4580fe187a806--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62c5a4a68cc4580fe187a806>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.153-alpha.4b5cb5.0<br/>@codecademy/gamut-labs@32.4.1-alpha.4b5cb5.0<br/>@codecademy/gamut@48.1.0-alpha.4b5cb5.0<br/>@codecademy/monaco@0.6.56-alpha.4b5cb5.0<br/>@codecademy/styleguide@58.2.0-alpha.4b5cb5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62c5ab34274ef41722cd9a8a--gamut-preview.netlify.app>https://62c5ab34274ef41722cd9a8a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62c5ab34274ef41722cd9a8a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.153-alpha.45d5c6.0<br/>@codecademy/gamut-labs@32.4.1-alpha.45d5c6.0<br/>@codecademy/gamut@48.1.0-alpha.45d5c6.0<br/>@codecademy/monaco@0.6.56-alpha.45d5c6.0<br/>@codecademy/styleguide@58.2.0-alpha.45d5c6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62c5add6355c3018d1ab6202--gamut-preview.netlify.app>https://62c5add6355c3018d1ab6202--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62c5add6355c3018d1ab6202>Deploy Logs</a></p>

File: packages/gamut/src/Modals/Modal.tsx [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
+  href?: string;
+  onClick?: ComponentProps<typeof FillButton>['onClick'];
+  disabled?: boolean;
+}
+interface ModalView {
+  title?: string;
+  children: React.ReactNode;
+  nextCta?: DialogButtonProps;
+  confirmCta?: DialogButtonProps;
+  cancelCta?: DialogButtonProps;
```
Comment (saghdaey) [inline packages/gamut/src/Modals/Modal.tsx:22]: should any more of these be required?

**Review by sooinc — APPROVED** [review]
Looks great to me! 🙌

**Review by dreamwasp — COMMENTED** [review]
this is awesome!! my only concern is what happens if someone passes `children` to the Modal *and* views? i wonder if we could make this a [discriminated union ](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types) type to keep that from happening

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.164-alpha.fc5532.0<br/>@codecademy/gamut-labs@32.7.7-alpha.fc5532.0<br/>@codecademy/gamut@48.3.0-alpha.fc5532.0<br/>@codecademy/monaco@0.6.59-alpha.fc5532.0<br/>@codecademy/styleguide@58.6.0-alpha.fc5532.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62d72663b37fce29836ee9d9--gamut-preview.netlify.app>https://62d72663b37fce29836ee9d9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62d72663b37fce29836ee9d9>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
tiny thing but looks awesome!

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62d80fecb335a1603ef85eb9--gamut-preview.netlify.app>https://62d80fecb335a1603ef85eb9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62d80fecb335a1603ef85eb9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.164-alpha.ba27f8.0<br/>@codecademy/gamut-labs@32.7.7-alpha.ba27f8.0<br/>@codecademy/gamut@48.3.0-alpha.ba27f8.0<br/>@codecademy/monaco@0.6.59-alpha.ba27f8.0<br/>@codecademy/styleguide@58.6.0-alpha.ba27f8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62d81888f8cea603300de14d--gamut-preview.netlify.app>https://62d81888f8cea603300de14d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62d81888f8cea603300de14d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.164-alpha.e0701e.0<br/>@codecademy/gamut-labs@32.7.7-alpha.e0701e.0<br/>@codecademy/gamut@48.3.0-alpha.e0701e.0<br/>@codecademy/monaco@0.6.59-alpha.e0701e.0<br/>@codecademy/styleguide@58.6.0-alpha.e0701e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62d81f6c3e5ae30b7193e4bb--gamut-preview.netlify.app>https://62d81f6c3e5ae30b7193e4bb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62d81f6c3e5ae30b7193e4bb>Deploy Logs</a></p>

**Pull request merged**